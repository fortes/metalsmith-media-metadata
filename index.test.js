/* eslint-env node,jest */
const {promisify} = require('util');
const mediaMetadata = require('./index');
const Metalsmith = require('metalsmith');
const path = require('path');
const rimraf = promisify(require('rimraf'));

let files;
let metal;

beforeAll(done => {
  // Setup metalsmith, but don't build yet.
  metal = Metalsmith(__dirname)
    .source('./')
    .ignore(['.*', 'node_modules', '__*'])
    .destination('__build')
    .use(mediaMetadata({}));

  // .use(mediaMetadata({}))
  metal.build((err, processedFiles) => {
    files = processedFiles;
    done(err);
  });
});

afterAll(() => {
  return rimraf(path.join(__dirname, '__build'));
});

it('adds exif data to images', () => {
  // If build pipeline fails, `files` never gets assigned
  expect(files).toBeDefined();

  // Exiftool output varies slightly by version, etc. Pick just a few fields to
  // check on.
  expect(files['sample.jpg'].exif).toBeDefined();
  expect(files['sample.jpg'].exif.ImageWidth).toBe(640);
  expect(files['sample.jpg'].exif.Flash).toBe('Off, Did not fire');
  expect(files['sample2.jpg'].exif).toBeDefined();
  expect(files['sample2.jpg'].exif.AFAreaMode).toBe('Single Area');
  expect(files['sample2.jpg'].exif.GPSLongitude).toBe(`11 deg 52' 53.32" E`);
});

it('adds exif data to video', () => {
  // If build pipeline fails, `files` never gets assigned
  expect(files).toBeDefined();

  const data = files['test-mpeg_512kb.mp4'];
  expect(data.exif).toBeDefined();

  expect(data.exif.TrackDuration).toBe('21.00 s');
  expect(data.exif.AudioChannels).toBe(2);
});

it('ignores non-media files', () => {
  expect(files).toBeDefined();
  expect(files['README.md'].exif).toBeUndefined();
});
