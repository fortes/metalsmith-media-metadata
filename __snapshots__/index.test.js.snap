// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds exif data to images 1`] = `
Object {
  "AFAreaMode": "Single Area",
  "AFPoint": "Center",
  "AFPointsInFocus": "(none)",
  "ActiveD-Lighting": "Off",
  "Aperture": 5.9,
  "AuxiliaryLens": "Off",
  "BitsPerSample": 8,
  "Brightness": "n/a",
  "CircleOfConfusion": "0.006 mm",
  "ColorComponents": 3,
  "ColorMode": "Color",
  "ColorSpace": "sRGB",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "Compression": "JPEG (old-style)",
  "Contrast": "Normal",
  "CreateDate": "2008:10:22 16:28:39",
  "CustomRendered": "Normal",
  "DataDump": "(Binary data 2542 bytes, use -b option to extract)",
  "DateTimeOriginal": "2008:10:22 16:28:39",
  "DigitalZoom": 1,
  "DigitalZoomRatio": 0,
  "Directory": "/home/fortes/metalsmith-media-metadata",
  "DistortionControl": "Off",
  "EncodingProcess": "Baseline DCT, Huffman coding",
  "ExifByteOrder": "Little-endian (Intel, II)",
  "ExifImageHeight": 480,
  "ExifImageWidth": 640,
  "ExifToolVersion": 10.4,
  "ExifVersion": "0220",
  "ExposureCompensation": 0,
  "ExposureMode": "Auto",
  "ExposureProgram": "Program AE",
  "ExposureTime": "1/75",
  "FNumber": 5.9,
  "FOV": "18.3 deg",
  "FaceDetectFrameSize": "320 240",
  "FacesDetected": 0,
  "FileAccessDate": "2017:09:18 03:34:11-07:00",
  "FileInodeChangeDate": "2017:09:18 09:29:43-07:00",
  "FileModifyDate": "2017:09:18 03:34:11-07:00",
  "FileName": "sample.jpg",
  "FilePermissions": "rw-r--r--",
  "FileSize": "158 kB",
  "FileSource": "Digital Camera",
  "FileType": "JPEG",
  "FileTypeExtension": "jpg",
  "FilterEffect": "n/a",
  "Flash": "Off, Did not fire",
  "FlashSetting": "",
  "FlashpixVersion": "0100",
  "FocalLength": "24.0 mm",
  "FocalLength35efl": "24.0 mm (35 mm equivalent: 112.0 mm)",
  "FocalLengthIn35mmFormat": "112 mm",
  "FocusMode": "AF-S",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSDateStamp": "2008:10:23",
  "GPSDateTime": "2008:10:23 14:27:07.24Z",
  "GPSImgDirectionRef": "Unknown ()",
  "GPSLatitude": "43 deg 28' 2.81\\" N",
  "GPSLatitudeRef": "North",
  "GPSLongitude": "11 deg 53' 6.46\\" E",
  "GPSLongitudeRef": "East",
  "GPSMapDatum": "WGS-84   ",
  "GPSPosition": "43 deg 28' 2.81\\" N, 11 deg 53' 6.46\\" E",
  "GPSSatellites": "06",
  "GPSTimeStamp": "14:27:07.24",
  "GainControl": "None",
  "HueAdjustment": "n/a",
  "HyperfocalDistance": "15.16 m",
  "ISO": 64,
  "ISOSelection": "Auto",
  "ImageAdjustment": "Normal",
  "ImageDescription": "                               ",
  "ImageHeight": 480,
  "ImageSize": "640x480",
  "ImageStabilization": "VR-On",
  "ImageWidth": 640,
  "InteropIndex": "R98 - DCF basic file (sRGB)",
  "InteropVersion": "0100",
  "LightSource": "Unknown",
  "LightValue": 12,
  "MIMEType": "image/jpeg",
  "Make": "NIKON",
  "MakerNoteVersion": 2.1,
  "MaxApertureValue": 2.7,
  "Megapixels": 0.307,
  "MeteringMode": "Multi-segment",
  "Model": "COOLPIX P6000",
  "ModifyDate": "2008:11:01 21:15:07",
  "NEFBitDepth": "n/a (JPEG)",
  "NikonCaptureVersion": "COOLPIX P6000V1.0              ",
  "NoiseReduction": "Off",
  "Orientation": "Horizontal (normal)",
  "PictureControlAdjust": "Default Settings",
  "PictureControlBase": "Standard",
  "PictureControlName": "Standard",
  "PictureControlQuickAdjust": "Normal",
  "PictureControlVersion": "0100",
  "Quality": "Fine",
  "RatingPercent": 0,
  "ResolutionUnit": "inches",
  "RetouchHistory": "None",
  "Saturation": "Normal",
  "ScaleFactor35efl": 4.7,
  "SceneAssist": "",
  "SceneCaptureType": "Standard",
  "SceneMode": "",
  "SceneType": "Directly photographed",
  "Sharpness": "Normal",
  "ShutterSpeed": "1/75",
  "Software": "Nikon Transfer 1.1 W",
  "SourceFile": "/home/fortes/metalsmith-media-metadata/sample.jpg",
  "SubjectDistanceRange": "Unknown",
  "ThumbnailImage": "(Binary data 6702 bytes, use -b option to extract)",
  "ThumbnailLength": 6702,
  "ThumbnailOffset": 4560,
  "ToneComp": "Normal",
  "ToningEffect": "n/a",
  "ToningSaturation": "n/a",
  "UserComment": "",
  "WhiteBalance": "Auto",
  "XMPToolkit": "Public XMP Toolkit Core 3.5",
  "XResolution": 300,
  "YCbCrPositioning": "Centered",
  "YCbCrSubSampling": "YCbCr4:2:2 (2 1)",
  "YResolution": 300,
}
`;

exports[`adds exif data to images 2`] = `
Object {
  "AFAreaMode": "Single Area",
  "AFPoint": "Center",
  "AFPointsInFocus": "(none)",
  "ActiveD-Lighting": "Off",
  "Aperture": 4.4,
  "AuxiliaryLens": "Off",
  "BitsPerSample": 8,
  "Brightness": "n/a",
  "CircleOfConfusion": "0.006 mm",
  "ColorComponents": 3,
  "ColorMode": "Color",
  "ColorSpace": "sRGB",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "Compression": "JPEG (old-style)",
  "Contrast": "Normal",
  "CreateDate": "2008:10:22 17:00:07",
  "CustomRendered": "Normal",
  "DataDump": "(Binary data 2542 bytes, use -b option to extract)",
  "DateTimeOriginal": "2008:10:22 17:00:07",
  "DigitalZoom": 1,
  "DigitalZoomRatio": 0,
  "Directory": "/home/fortes/metalsmith-media-metadata",
  "DistortionControl": "Off",
  "EncodingProcess": "Baseline DCT, Huffman coding",
  "ExifByteOrder": "Little-endian (Intel, II)",
  "ExifImageHeight": 480,
  "ExifImageWidth": 640,
  "ExifToolVersion": 10.4,
  "ExifVersion": "0220",
  "ExposureCompensation": 0,
  "ExposureMode": "Auto",
  "ExposureProgram": "Program AE",
  "ExposureTime": "1/100",
  "FNumber": 4.4,
  "FOV": "28.8 deg",
  "FaceDetectFrameSize": "320 240",
  "FacesDetected": 0,
  "FileAccessDate": "2017:09:18 03:34:36-07:00",
  "FileInodeChangeDate": "2017:09:18 09:29:43-07:00",
  "FileModifyDate": "2017:09:18 03:34:36-07:00",
  "FileName": "sample2.jpg",
  "FilePermissions": "rw-r--r--",
  "FileSize": "153 kB",
  "FileSource": "Digital Camera",
  "FileType": "JPEG",
  "FileTypeExtension": "jpg",
  "FilterEffect": "n/a",
  "Flash": "Off, Did not fire",
  "FlashSetting": "",
  "FlashpixVersion": "0100",
  "FocalLength": "15.0 mm",
  "FocalLength35efl": "15.0 mm (35 mm equivalent: 70.0 mm)",
  "FocalLengthIn35mmFormat": "70 mm",
  "FocusMode": "AF-S",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSDateStamp": "2008:10:23",
  "GPSDateTime": "2008:10:23 14:57:41.37Z",
  "GPSImgDirectionRef": "Unknown ()",
  "GPSLatitude": "43 deg 27' 52.04\\" N",
  "GPSLatitudeRef": "North",
  "GPSLongitude": "11 deg 52' 53.32\\" E",
  "GPSLongitudeRef": "East",
  "GPSMapDatum": "WGS-84   ",
  "GPSPosition": "43 deg 27' 52.04\\" N, 11 deg 52' 53.32\\" E",
  "GPSSatellites": "04",
  "GPSTimeStamp": "14:57:41.37",
  "GainControl": "None",
  "HueAdjustment": "n/a",
  "HyperfocalDistance": "7.94 m",
  "ISO": 64,
  "ISOSelection": "Auto",
  "ImageAdjustment": "Normal",
  "ImageDescription": "                               ",
  "ImageHeight": 480,
  "ImageSize": "640x480",
  "ImageStabilization": "VR-On",
  "ImageWidth": 640,
  "InteropIndex": "R98 - DCF basic file (sRGB)",
  "InteropVersion": "0100",
  "LightSource": "Unknown",
  "LightValue": 11.6,
  "MIMEType": "image/jpeg",
  "Make": "NIKON",
  "MakerNoteVersion": 2.1,
  "MaxApertureValue": 2.7,
  "Megapixels": 0.307,
  "MeteringMode": "Multi-segment",
  "Model": "COOLPIX P6000",
  "ModifyDate": "2008:11:01 21:15:11",
  "NEFBitDepth": "n/a (JPEG)",
  "NikonCaptureVersion": "COOLPIX P6000V1.0              ",
  "NoiseReduction": "Off",
  "Orientation": "Horizontal (normal)",
  "PictureControlAdjust": "Full Control",
  "PictureControlBase": "Standard",
  "PictureControlName": "Standard",
  "PictureControlQuickAdjust": "+2",
  "PictureControlVersion": "0100",
  "Quality": "Fine",
  "RatingPercent": 0,
  "ResolutionUnit": "inches",
  "RetouchHistory": "None",
  "Saturation": "Normal",
  "ScaleFactor35efl": 4.7,
  "SceneAssist": "",
  "SceneCaptureType": "Standard",
  "SceneMode": "",
  "SceneType": "Directly photographed",
  "Sharpness": "Normal",
  "ShutterSpeed": "1/100",
  "Software": "Nikon Transfer 1.1 W",
  "SourceFile": "/home/fortes/metalsmith-media-metadata/sample2.jpg",
  "SubjectDistanceRange": "Unknown",
  "ThumbnailImage": "(Binary data 6474 bytes, use -b option to extract)",
  "ThumbnailLength": 6474,
  "ThumbnailOffset": 4560,
  "ToneComp": "Normal",
  "ToningEffect": "n/a",
  "ToningSaturation": "n/a",
  "UserComment": "",
  "WhiteBalance": "Auto",
  "XMPToolkit": "Public XMP Toolkit Core 3.5",
  "XResolution": 300,
  "YCbCrPositioning": "Centered",
  "YCbCrSubSampling": "YCbCr4:2:2 (2 1)",
  "YResolution": 300,
}
`;
